File: tinysteps_flutter/lib/models/interests.dart
class InterestCategory {
  final String id;
  final String name;
  final String emoji;
  final List<Interest> interests;

  const InterestCategory({
    required this.id,
    required this.name,
    required this.emoji,
    required this.interests,
  });
}

class Interest {
  final String id;
  final String name;
  final String emoji;
  final String category;

  const Interest({
    required this.id,
    required this.name,
    required this.emoji,
    required this.category,
  });
}

class InterestData {
  static const List<InterestCategory> categories = [
    InterestCategory(
      id: 'animals',
      name: 'Animals',
      emoji: '🐾',
      interests: [
        Interest(id: 'dinosaurs', name: 'Dinosaurs', emoji: '🦕', category: 'animals'),
        Interest(id: 'dogs', name: 'Dogs', emoji: '🐕', category: 'animals'),
        Interest(id: 'cats', name: 'Cats', emoji: '🐱', category: 'animals'),
        Interest(id: 'birds', name: 'Birds', emoji: '🐦', category: 'animals'),
        Interest(id: 'fish', name: 'Fish', emoji: '🐠', category: 'animals'),
        Interest(id: 'horses', name: 'Horses', emoji: '🐴', category: 'animals'),
        Interest(id: 'elephants', name: 'Elephants', emoji: '🐘', category: 'animals'),
        Interest(id: 'lions', name: 'Lions', emoji: '🦁', category: 'animals'),
        Interest(id: 'butterflies', name: 'Butterflies', emoji: '🦋', category: 'animals'),
        Interest(id: 'bugs', name: 'Bugs', emoji: '🐛', category: 'animals'),
      ],
    ),
    InterestCategory(
      id: 'vehicles',
      name: 'Vehicles',
      emoji: '🚗',
      interests: [
        Interest(id: 'cars', name: 'Cars', emoji: '🚗', category: 'vehicles'),
        Interest(id: 'trucks', name: 'Trucks', emoji: '🚚', category: 'vehicles'),
        Interest(id: 'trains', name: 'Trains', emoji: '🚂', category: 'vehicles'),
        Interest(id: 'planes', name: 'Planes', emoji: '✈️', category: 'vehicles'),
        Interest(id: 'boats', name: 'Boats', emoji: '⛵', category: 'vehicles'),
        Interest(id: 'rockets', name: 'Rockets', emoji: '🚀', category: 'vehicles'),
        Interest(id: 'tractors', name: 'Tractors', emoji: '🚜', category: 'vehicles'),
        Interest(id: 'helicopters', name: 'Helicopters', emoji: '🚁', category: 'vehicles'),
      ],
    ),
    InterestCategory(
      id: 'activities',
      name: 'Activities',
      emoji: '⚽',
      interests: [
        Interest(id: 'dancing', name: 'Dancing', emoji: '💃', category: 'activities'),
        Interest(id: 'singing', name: 'Singing', emoji: '🎤', category: 'activities'),
        Interest(id: 'drawing', name: 'Drawing', emoji: '🎨', category: 'activities'),
        Interest(id: 'building', name: 'Building', emoji: '🧱', category: 'activities'),
        Interest(id: 'reading', name: 'Reading', emoji: '📖', category: 'activities'),
        Interest(id: 'sports', name: 'Sports', emoji: '⚽', category: 'activities'),
        Interest(id: 'cooking', name: 'Cooking', emoji: '👨‍🍳', category: 'activities'),
        Interest(id: 'gardening', name: 'Gardening', emoji: '🌱', category: 'activities'),
      ],
    ),
    InterestCategory(
      id: 'nature',
      name: 'Nature',
      emoji: '🌿',
      interests: [
        Interest(id: 'flowers', name: 'Flowers', emoji: '🌸', category: 'nature'),
        Interest(id: 'trees', name: 'Trees', emoji: '🌳', category: 'nature'),
        Interest(id: 'ocean', name: 'Ocean', emoji: '🌊', category: 'nature'),
        Interest(id: 'mountains', name: 'Mountains', emoji: '⛰️', category: 'nature'),
        Interest(id: 'stars', name: 'Stars', emoji: '⭐', category: 'nature'),
        Interest(id: 'rainbows', name: 'Rainbows', emoji: '🌈', category: 'nature'),
        Interest(id: 'weather', name: 'Weather', emoji: '☀️', category: 'nature'),
      ],
    ),
    InterestCategory(
      id: 'fantasy',
      name: 'Fantasy',
      emoji: '✨',
      interests: [
        Interest(id: 'princesses', name: 'Princesses', emoji: '👸', category: 'fantasy'),
        Interest(id: 'superheroes', name: 'Superheroes', emoji: '🦸', category: 'fantasy'),
        Interest(id: 'dragons', name: 'Dragons', emoji: '🐉', category: 'fantasy'),
        Interest(id: 'unicorns', name: 'Unicorns', emoji: '🦄', category: 'fantasy'),
        Interest(id: 'fairies', name: 'Fairies', emoji: '🧚', category: 'fantasy'),
        Interest(id: 'pirates', name: 'Pirates', emoji: '🏴‍☠️', category: 'fantasy'),
        Interest(id: 'robots', name: 'Robots', emoji: '🤖', category: 'fantasy'),
        Interest(id: 'magic', name: 'Magic', emoji: '🪄', category: 'fantasy'),
      ],
    ),
    InterestCategory(
      id: 'food',
      name: 'Food',
      emoji: '🍎',
      interests: [
        Interest(id: 'fruits', name: 'Fruits', emoji: '🍎', category: 'food'),
        Interest(id: 'vegetables', name: 'Vegetables', emoji: '🥕', category: 'food'),
        Interest(id: 'sweets', name: 'Sweets', emoji: '🍪', category: 'food'),
        Interest(id: 'pizza', name: 'Pizza', emoji: '🍕', category: 'food'),
        Interest(id: 'icecream', name: 'Ice Cream', emoji: '🍦', category: 'food'),
      ],
    ),
    InterestCategory(
      id: 'music',
      name: 'Music',
      emoji: '🎵',
      interests: [
        Interest(id: 'nursery_rhymes', name: 'Nursery Rhymes', emoji: '🎶', category: 'music'),
        Interest(id: 'instruments', name: 'Instruments', emoji: '🎸', category: 'music'),
        Interest(id: 'drums', name: 'Drums', emoji: '🥁', category: 'music'),
        Interest(id: 'piano', name: 'Piano', emoji: '🎹', category: 'music'),
      ],
    ),
    InterestCategory(
      id: 'seasons',
      name: 'Seasons',
      emoji: '🍂',
      interests: [
        Interest(id: 'spring', name: 'Spring', emoji: '🌷', category: 'seasons'),
        Interest(id: 'summer', name: 'Summer', emoji: '☀️', category: 'seasons'),
        Interest(id: 'autumn', name: 'Autumn', emoji: '🍂', category: 'seasons'),
        Interest(id: 'winter', name: 'Winter', emoji: '❄️', category: 'seasons'),
        Interest(id: 'holidays', name: 'Holidays', emoji: '🎄', category: 'seasons'),
      ],
    ),
    InterestCategory(
      id: 'characters',
      name: 'Characters',
      emoji: '🧸',
      interests: [
        Interest(id: 'teddy_bears', name: 'Teddy Bears', emoji: '🧸', category: 'characters'),
        Interest(id: 'dolls', name: 'Dolls', emoji: '🪆', category: 'characters'),
        Interest(id: 'action_figures', name: 'Action Figures', emoji: '🦸‍♂️', category: 'characters'),
      ],
    ),
  ];

  static List<Interest> get allInterests {
    return categories.expand((cat) => cat.interests).toList();
  }

  static Interest? getInterestById(String id) {
    for (final category in categories) {
      for (final interest in category.interests) {
        if (interest.id == id) {
          return interest;
        }
      }
    }
    return null;
  }
}

class FavoriteColor {
  final String id;
  final String name;
  final String hexCode;

  const FavoriteColor({
    required this.id,
    required this.name,
    required this.hexCode,
  });

  static const List<FavoriteColor> colors = [
    FavoriteColor(id: 'red', name: 'Red', hexCode: '#EF4444'),
    FavoriteColor(id: 'orange', name: 'Orange', hexCode: '#F97316'),
    FavoriteColor(id: 'yellow', name: 'Yellow', hexCode: '#EAB308'),
    FavoriteColor(id: 'green', name: 'Green', hexCode: '#22C55E'),
    FavoriteColor(id: 'blue', name: 'Blue', hexCode: '#3B82F6'),
    FavoriteColor(id: 'purple', name: 'Purple', hexCode: '#A855F7'),
    FavoriteColor(id: 'pink', name: 'Pink', hexCode: '#EC4899'),
    FavoriteColor(id: 'teal', name: 'Teal', hexCode: '#14B8A6'),
    FavoriteColor(id: 'indigo', name: 'Indigo', hexCode: '#6366F1'),
    FavoriteColor(id: 'brown', name: 'Brown', hexCode: '#A16207'),
    FavoriteColor(id: 'gray', name: 'Gray', hexCode: '#6B7280'),
    FavoriteColor(id: 'black', name: 'Black', hexCode: '#1F2937'),
  ];
}

class FavoriteCharacter {
  final String id;
  final String name;
  final String? franchise;

  const FavoriteCharacter({
    required this.id,
    required this.name,
    this.franchise,
  });

  static const List<FavoriteCharacter> popularCharacters = [
    FavoriteCharacter(id: 'peppa_pig', name: 'Peppa Pig', franchise: 'Peppa Pig'),
    FavoriteCharacter(id: 'paw_patrol', name: 'Paw Patrol', franchise: 'Paw Patrol'),
    FavoriteCharacter(id: 'cocomelon', name: 'CoComelon', franchise: 'CoComelon'),
    FavoriteCharacter(id: 'bluey', name: 'Bluey', franchise: 'Bluey'),
    FavoriteCharacter(id: 'elsa', name: 'Elsa', franchise: 'Frozen'),
    FavoriteCharacter(id: 'spiderman', name: 'Spider-Man', franchise: 'Marvel'),
    FavoriteCharacter(id: 'mickey', name: 'Mickey Mouse', franchise: 'Disney'),
    FavoriteCharacter(id: 'minnie', name: 'Minnie Mouse', franchise: 'Disney'),
    FavoriteCharacter(id: 'thomas', name: 'Thomas', franchise: 'Thomas & Friends'),
    FavoriteCharacter(id: 'dora', name: 'Dora', franchise: 'Dora the Explorer'),
    FavoriteCharacter(id: 'winnie', name: 'Winnie the Pooh', franchise: 'Disney'),
    FavoriteCharacter(id: 'elmo', name: 'Elmo', franchise: 'Sesame Street'),
    FavoriteCharacter(id: 'lightning', name: 'Lightning McQueen', franchise: 'Cars'),
    FavoriteCharacter(id: 'moana', name: 'Moana', franchise: 'Disney'),
    FavoriteCharacter(id: 'batman', name: 'Batman', franchise: 'DC'),
    FavoriteCharacter(id: 'hello_kitty', name: 'Hello Kitty', franchise: 'Sanrio'),
  ];
}

File: tinysteps-ai/data/interests.ts
import { Interest } from '../types';

export const AVAILABLE_INTERESTS: Interest[] = [
  // Animals
  { id: 'dinosaurs', name: 'Dinosaurs', icon: '🦕', category: 'animals' },
  { id: 'dogs', name: 'Dogs', icon: '🐕', category: 'animals' },
  { id: 'cats', name: 'Cats', icon: '🐱', category: 'animals' },
  { id: 'horses', name: 'Horses', icon: '🐴', category: 'animals' },
  { id: 'birds', name: 'Birds', icon: '🐦', category: 'animals' },
  { id: 'fish', name: 'Fish', icon: '🐠', category: 'animals' },
  { id: 'butterflies', name: 'Butterflies', icon: '🦋', category: 'animals' },
  { id: 'bears', name: 'Bears', icon: '🐻', category: 'animals' },
  { id: 'elephants', name: 'Elephants', icon: '🐘', category: 'animals' },
  { id: 'lions', name: 'Lions', icon: '🦁', category: 'animals' },
  { id: 'bunnies', name: 'Bunnies', icon: '🐰', category: 'animals' },
  { id: 'pandas', name: 'Pandas', icon: '🐼', category: 'animals' },

  // Vehicles
  { id: 'cars', name: 'Cars', icon: '🚗', category: 'vehicles' },
  { id: 'trucks', name: 'Trucks', icon: '🚚', category: 'vehicles' },
  { id: 'trains', name: 'Trains', icon: '🚂', category: 'vehicles' },
  { id: 'planes', name: 'Planes', icon: '✈️', category: 'vehicles' },
  { id: 'boats', name: 'Boats', icon: '⛵', category: 'vehicles' },
  { id: 'rockets', name: 'Rockets', icon: '🚀', category: 'vehicles' },
  { id: 'tractors', name: 'Tractors', icon: '🚜', category: 'vehicles' },
  { id: 'helicopters', name: 'Helicopters', icon: '🚁', category: 'vehicles' },
  { id: 'motorcycles', name: 'Motorcycles', icon: '🏍️', category: 'vehicles' },
  { id: 'fire-trucks', name: 'Fire Trucks', icon: '🚒', category: 'vehicles' },

  // Nature
  { id: 'flowers', name: 'Flowers', icon: '🌸', category: 'nature' },
  { id: 'trees', name: 'Trees', icon: '🌳', category: 'nature' },
  { id: 'ocean', name: 'Ocean', icon: '🌊', category: 'nature' },
  { id: 'mountains', name: 'Mountains', icon: '⛰️', category: 'nature' },
  { id: 'stars', name: 'Stars', icon: '⭐', category: 'nature' },
  { id: 'moon', name: 'Moon', icon: '🌙', category: 'nature' },
  { id: 'rainbow', name: 'Rainbow', icon: '🌈', category: 'nature' },
  { id: 'sunshine', name: 'Sunshine', icon: '☀️', category: 'nature' },
  { id: 'rain', name: 'Rain', icon: '🌧️', category: 'nature' },
  { id: 'snow', name: 'Snow', icon: '❄️', category: 'nature' },

  // Fantasy
  { id: 'princesses', name: 'Princesses', icon: '👸', category: 'fantasy' },
  { id: 'superheroes', name: 'Superheroes', icon: '🦸', category: 'fantasy' },
  { id: 'dragons', name: 'Dragons', icon: '🐉', category: 'fantasy' },
  { id: 'unicorns', name: 'Unicorns', icon: '🦄', category: 'fantasy' },
  { id: 'fairies', name: 'Fairies', icon: '🧚', category: 'fantasy' },
  { id: 'wizards', name: 'Wizards', icon: '🧙', category: 'fantasy' },
  { id: 'pirates', name: 'Pirates', icon: '🏴‍☠️', category: 'fantasy' },
  { id: 'robots', name: 'Robots', icon: '🤖', category: 'fantasy' },
  { id: 'mermaids', name: 'Mermaids', icon: '🧜‍♀️', category: 'fantasy' },
  { id: 'knights', name: 'Knights', icon: '🛡️', category: 'fantasy' },

  // Sports
  { id: 'soccer', name: 'Soccer', icon: '⚽', category: 'sports' },
  { id: 'basketball', name: 'Basketball', icon: '🏀', category: 'sports' },
  { id: 'swimming', name: 'Swimming', icon: '🏊', category: 'sports' },
  { id: 'dancing', name: 'Dancing', icon: '💃', category: 'sports' },
  { id: 'running', name: 'Running', icon: '🏃', category: 'sports' },
  { id: 'cycling', name: 'Cycling', icon: '🚴', category: 'sports' },
  { id: 'gymnastics', name: 'Gymnastics', icon: '🤸', category: 'sports' },
  { id: 'tennis', name: 'Tennis', icon: '🎾', category: 'sports' },

  // Arts
  { id: 'painting', name: 'Painting', icon: '🎨', category: 'arts' },
  { id: 'drawing', name: 'Drawing', icon: '✏️', category: 'arts' },
  { id: 'crafts', name: 'Crafts', icon: '✂️', category: 'arts' },
  { id: 'clay', name: 'Play-Doh/Clay', icon: '🎭', category: 'arts' },
  { id: 'coloring', name: 'Coloring', icon: '🖍️', category: 'arts' },
  { id: 'building', name: 'Building', icon: '🧱', category: 'arts' },
  { id: 'puzzles', name: 'Puzzles', icon: '🧩', category: 'arts' },

  // Science
  { id: 'space', name: 'Space', icon: '🌌', category: 'science' },
  { id: 'bugs', name: 'Bugs & Insects', icon: '🐛', category: 'science' },
  { id: 'weather', name: 'Weather', icon: '🌤️', category: 'science' },
  { id: 'experiments', name: 'Experiments', icon: '🔬', category: 'science' },
  { id: 'numbers', name: 'Numbers', icon: '🔢', category: 'science' },
  { id: 'letters', name: 'Letters', icon: '🔤', category: 'science' },
  { id: 'magnets', name: 'Magnets', icon: '🧲', category: 'science' },

  // Music
  { id: 'singing', name: 'Singing', icon: '🎤', category: 'music' },
  { id: 'drums', name: 'Drums', icon: '🥁', category: 'music' },
  { id: 'piano', name: 'Piano', icon: '🎹', category: 'music' },
  { id: 'guitar', name: 'Guitar', icon: '🎸', category: 'music' },
  { id: 'nursery-rhymes', name: 'Nursery Rhymes', icon: '🎵', category: 'music' },

  // Other
  { id: 'cooking', name: 'Cooking', icon: '👨‍🍳', category: 'other' },
  { id: 'gardening', name: 'Gardening', icon: '🌱', category: 'other' },
  { id: 'reading', name: 'Reading', icon: '📚', category: 'other' },
  { id: 'dolls', name: 'Dolls', icon: '🪆', category: 'other' },
  { id: 'blocks', name: 'Blocks', icon: '🧊', category: 'other' },
  { id: 'balls', name: 'Balls', icon: '🏐', category: 'other' },
  { id: 'bubbles', name: 'Bubbles', icon: '🫧', category: 'other' },
  { id: 'sand', name: 'Sand Play', icon: '🏖️', category: 'other' },
  { id: 'water-play', name: 'Water Play', icon: '💧', category: 'other' },
];

export const INTEREST_CATEGORIES = [
  { id: 'animals', name: 'Animals', icon: '🐾' },
  { id: 'vehicles', name: 'Vehicles', icon: '🚗' },
  { id: 'nature', name: 'Nature', icon: '🌿' },
  { id: 'fantasy', name: 'Fantasy', icon: '✨' },
  { id: 'sports', name: 'Sports', icon: '🏃' },
  { id: 'arts', name: 'Arts & Crafts', icon: '🎨' },
  { id: 'science', name: 'Learning', icon: '🔬' },
  { id: 'music', name: 'Music', icon: '🎵' },
  { id: 'other', name: 'Other', icon: '🎮' },
];

export const POPULAR_CHARACTERS = [
  'Peppa Pig',
  'Bluey',
  'Paw Patrol',
  'Cocomelon',
  'Mickey Mouse',
  'Minnie Mouse',
  'Elsa (Frozen)',
  'Spider-Man',
  'Batman',
  'Dora the Explorer',
  'Thomas the Tank Engine',
  'Barbie',
  'Baby Shark',
  'Sesame Street',
  'Winnie the Pooh',
  'Finding Nemo',
  'Toy Story',
  'The Lion King',
  'Moana',
  'Cars (Lightning McQueen)',
];

export const FAVORITE_COLORS = [
  { name: 'Red', hex: '#EF4444' },
  { name: 'Blue', hex: '#3B82F6' },
  { name: 'Green', hex: '#22C55E' },
  { name: 'Yellow', hex: '#EAB308' },
  { name: 'Purple', hex: '#A855F7' },
  { name: 'Pink', hex: '#EC4899' },
  { name: 'Orange', hex: '#F97316' },
  { name: 'Teal', hex: '#14B8A6' },
  { name: 'Rainbow', hex: 'linear-gradient(90deg, red, orange, yellow, green, blue, purple)' },
];

export function getInterestsByCategory(category: string): Interest[] {
  return AVAILABLE_INTERESTS.filter((i) => i.category === category);
}

export function getInterestById(id: string): Interest | undefined {
  return AVAILABLE_INTERESTS.find((i) => i.id === id);
}

export function getPersonalizedGreeting(name: string, interests: Interest[]): string {
  if (interests.length === 0) {
    return `Hi there, ${name}!`;
  }

  const interest = interests[Math.floor(Math.random() * interests.length)];

  const greetings: Record<string, string[]> = {
    dinosaurs: [
      `ROAR! ${name} the mighty dinosaur explorer is here!`,
      `Ready for a dino-mite adventure, ${name}?`,
    ],
    unicorns: [
      `Magical greetings, ${name}!`,
      `Sparkle on, little ${name}!`,
    ],
    cars: [
      `Vroom vroom! ${name} is ready to roll!`,
      `Start your engines, ${name}!`,
    ],
    rockets: [
      `3, 2, 1... Blast off, ${name}!`,
      `Ready for lift-off, astronaut ${name}?`,
    ],
    superheroes: [
      `Super ${name} to the rescue!`,
      `Hero mode activated for ${name}!`,
    ],
    princesses: [
      `Welcome, Princess ${name}!`,
      `Your royal highness ${name} has arrived!`,
    ],
    default: [
      `Hey there, awesome ${name}!`,
      `${name} is ready for fun!`,
    ],
  };

  const options = greetings[interest.id] || greetings.default;
  return options[Math.floor(Math.random() * options.length)];
}

export function getThemedNotification(
  type: string,
  name: string,
  interests: Interest[]
): { title: string; message: string; icon: string } {
  const interest = interests.length > 0
    ? interests[Math.floor(Math.random() * interests.length)]
    : null;

  if (!interest) {
    return {
      title: 'Time for Fun!',
      message: `${name} has new activities to explore!`,
      icon: '🌟',
    };
  }

  const themes: Record<string, { title: string; message: string }> = {
    dinosaurs: {
      title: 'Dino Discovery!',
      message: `${name}, the dinosaurs have a new adventure waiting!`,
    },
    unicorns: {
      title: 'Magical Moment!',
      message: `${name}, sprinkle some magic on your day!`,
    },
    rockets: {
      title: 'Mission Control!',
      message: `Astronaut ${name}, you have a new mission!`,
    },
    cars: {
      title: 'Pit Stop Time!',
      message: `${name}, rev up for some exciting activities!`,
    },
  };

  const themed = themes[interest.id] || {
    title: `${interest.name} Time!`,
    message: `${name} has something fun to discover!`,
  };

  return {
    ...themed,
    icon: interest.icon,
  };
}
